File: haskell/Person.hs
class Alive a where
  mate :: a -> a -> a
  age :: a -> Int
  name :: a -> String
  gender :: a -> Gender

data Gender = Male | Female deriving (Show, Eq, Ord)

data Person = Person String Int Gender deriving (Show)

instance Alive Person where
  mate m n = (Person ((name m) ++ "-" ++ (name n)) 0 (gender n))
  age (Person _ a _) = a
  name (Person n _ _) = n
  gender (Person _ _ g) = g

parker = Person "Parker" 19 Male
tina = Person "Tina" 18 Female

File: c-c++/person.c
#include <stdio.h>

struct person
{
  char *name;
  int age;
};


void bday(struct person *p)
{
  p->age++;
}

void p_info(struct person *p)
{
  printf("%s's age is %d\n", p->name, p->age);
}

int main()
{
  struct person parker;
  parker.name = "Parker";
  parker.age = 19;

  p_info(&parker);
  
  printf("birthday!\n");
  bday(&parker);
  
  p_info(&parker);
  
  return 0;
}
